### PR TITLE
Replace Shapeshift API with native RPC searchTx

### PR DESCRIPTION
**Pull request by peachbits**
### CHANGELOG

Does this branch warrant an entry to the CHANGELOG?

- [ ] Yes
- [ ] No

### Dependencies

 none

### Description

 none

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205930238457636

### CODE REVIEW

File: src/cosmos/CosmosEngine.ts [hunk 1]
```diff
@@ -218,44 +194,143 @@ export class CosmosEngine extends CurrencyEngine<
...
+      networkFeeCoin = safeAddCoins([networkFeeCoin, ...amount])
+    }
+    const networkFee = networkFeeCoin.amount
+
+    const { coin, recipient, sender } = event
+
+    if (
+      this.walletInfo.keys.bech32Address !== recipient &&
+      this.walletInfo.keys.bech32Address !== sender &&
+      this.currencyInfo.currencyCode !== coin.denom.toUpperCase()
```
Comment (paullinator) [inline src/cosmos/CosmosEngine.ts:302]: if currencycode does match, but the address doesn't match as sender or recipient, why would we include this tx?

File: src/cosmos/CosmosEngine.ts [hunk 2]
```diff
@@ -218,44 +194,143 @@ export class CosmosEngine extends CurrencyEngine<
...
+
+          const block = await client.getBlock(tx.height)
+          const date = toSeconds(
+            fromRfc3339WithNanoseconds(block.header.time)
+          ).seconds
+          transferEvents.forEach(event => {
+            this.processCosmosTransaction(txidHex, date, event, tx)
+          })
+        }
+      } while (txCountTotal != null && txCountTotal > txCount && !earlyExit)
```
Comment (paullinator) [inline src/cosmos/CosmosEngine.ts:268]: `txCountTotal` can never be null by this point.

**Review by paullinator — CHANGES_REQUESTED** [review]

File: src/cosmos/CosmosEngine.ts [hunk 3]
```diff
@@ -218,44 +194,143 @@ export class CosmosEngine extends CurrencyEngine<
...
-    const { from, value } = message
+  async queryTransactionsInner(
+    queryString: typeof txQueryStrings[number]
+  ): Promise<string | undefined> {
+    const client = await this.getStargateClient()
 
-    const { amount: networkFee } = fee
+    // Using the tendermint client directly allows us to control the paging
+    // eslint-disable-next-line @typescript-eslint/dot-notation
+    const tmClient = client['forceGetTmClient']()
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/cosmos/CosmosEngine.ts:206]: It's protected so I need to suppress the error. I can use this or dot notation with ts-expect-error

File: src/cosmos/CosmosEngine.ts [hunk 4]
```diff
@@ -218,44 +194,143 @@ export class CosmosEngine extends CurrencyEngine<
...
 
-    const currencyCode = value.denom.toUpperCase()
-    let nativeAmount = value.amount
+          // The bank module emits 'transfer' events with sender, recipient, and coins
+          const transferEvents: TransferEvent[] = []
+          const parsedLogs = parseRawLog(tx.result.log)
+          parsedLogs.forEach(log => {
+            log.events.forEach(event => {
+              const transferEvent = asMaybe(asTransfer)(event)
+              if (transferEvent == null) return []
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/cosmos/CosmosEngine.ts:245]: remnant from using .map

File: src/cosmos/CosmosEngine.ts [hunk 5]
```diff
@@ -218,44 +194,143 @@ export class CosmosEngine extends CurrencyEngine<
...
+              const [recipient, sender, amount] = transferEvent.attributes
+
+              const coins = parseCoins(amount.value)
+              return coins.forEach(coin => {
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/cosmos/CosmosEngine.ts:249]: remnant from using .map

**Review by paullinator — APPROVED** [review]

**Pull request merged**